### ISSUE TITLE
[Feature RequestSupport for M1 Apple Silicon / arm64 architecture

### ISSUE THREAD

**Issue by gustavoabel** [googlesamples/mlkit#810]
**What's your feature request? Please describe.**
Please support MLKit on M1 Simulator
Please support it to run on a simulator for VisionOS which runs on M1 chip. Physical device won't be available soon
**Mobile environment**
iOS

**Additional context**

**Comment by gustavoabel** [comment]
@miworking3 Anything planned for this year?

**Comment by RogVok** [comment]
Looking at this comment: https://github.com/googlesamples/mlkit/issues/691#issuecomment-1625598912 this won't be addressed soon. 

P.S. Since Julie would be in this thread, I'd like to highlight that many of us are still awaiting the ability to include MLKit models directly into the app's bundle, rather than requiring a separate download on each device (Digital Ink). This would greatly enhance user experience and convenience. (and will boost the satisfaction of working with Google MLKit)

**Comment by evandorn** [comment]
+1 I have to use Google MLKit for a project and I'd really like to compile my app natively for simulator on my ARM Mac.

**Comment by importRyan** [comment]
+1 When camera and GPU resources are unavailable, as in a Simulator, Apple's throw errors or exceptions, but do compile. It would be a massive win if MLKit and MLImage adopted the same philosophy as a shortcut to getting this arm64 arch exclusion removed.

For context, this is a blocker for us adopting a few CI vendors' technologies to automate our snapshot testing, so it's not just a save-some-time-compiling situation.

**Comment by importRyan** [comment]
Any visitors, also see: https://issuetracker.google.com/issues/178965151 and join in several years of commentary. :)

**Comment by RutgerVleuten** [comment]
**+1** also bumping into the same issue.

**Comment by timmyjose** [comment]
This has made building my app impossible. Any idea when this would be supported?

**Comment by iwater** [comment]
+1

**Comment by angarsky** [comment]
+1

**Comment by yahyaghani** [comment]
+1

**Comment by Takuro-Researcher** [comment]
+1

**Comment by hanrw** [comment]
+1

**Comment by ronie-san** [comment]
+1

**Comment by Karllas** [comment]
+1

**Comment by majdsabahmaf** [comment]
As workaround if you are using Cocoapods, you can have different target that do not use the MLKit pod inside Podfile. and in your code also turn of all the code that depends on it when you use that target by using #if Release #endif for example.

Disadvantage: You will need to add any new file to new target and old target(s)

**Comment by tomas-dorda** [comment]
+1

**Comment by adamthewilliam** [comment]
+1

**Comment by batuhanseckinub** [comment]
+1

**Comment by Aranoledur** [comment]
Looks like Apple dropped support for x86 simulators in iOS26 so I simply can't build my project for iOS26 simulator. You can see in screenshots that I have iOS26 simulators, but they are not available as run destinations

![Image](https://github.com/user-attachments/assets/85c97ed7-f893-4de9-bc3c-d69cfa62f382)

![Image](https://github.com/user-attachments/assets/5cbf6c03-e5e4-4047-ac3a-1b3026ae1644)

**Comment by ualtinok** [comment]
+1

**Comment by thegreekconnect** [comment]
+1

**Comment by Jonny-P** [comment]
+1

I'd like to highlight that many of us are still awaiting the ability to include MLKit models directly into the app's bundle, rather than requiring a separate download on each device (Digital Ink). This would greatly enhance user experience and convenience. (and will boost the satisfaction of working with Google MLKit)

**Comment by twsl** [comment]
+1

**Comment by kae1616** [comment]
+1

**Comment by elbelga** [comment]
+1 🙏

**Comment by alamothe** [comment]
Google sucks

**Comment by mybosabhi** [comment]
+1

**Comment by sarim987** [comment]
+1

**Comment by nthtrung09it** [comment]
+1

**Comment by gotvitch** [comment]
+1

**Comment by ppamorim** [comment]
Please fix that, we can't run our application on iOS 26 simulator.

**Comment by AlixH** [comment]
Up

**Comment by nthtrung09it** [comment]
+1

**Comment by evandorn** [comment]
It's been 6 years since Apple Silicon came out. Nobody is even assigned to this issue. I think it's fair to say Google is not updating this and is likely no longer maintaining this project.

**Comment by elmsec** [comment]
For those who needs a workaround for testing on Xcode Simulators (arm64), you can use **Universal** version of Simulator. 

1. Remove the existing iOS platform runtime:
```
sudo rm -rf "/Library/Developer/CoreSimulator/Profiles/Runtimes/iOS <version>.simruntime"
```
_OR use `Xcode settings -> Components -> e.g. iOS 26.5 (delete this)`_

2. Install universal version instead:
```
xcodebuild -downloadPlatform iOS -architectureVariant universal
```

3. Restart your Mac or at least try this:
```
killall Simulator 2>/dev/null 
```

----
Note that using this Rosetta based universal version degrades the Simulator performance.

**Comment by gustavoabel** [comment]
> For those who needs a workaround for testing on Xcode Simulators (arm64), you can use **Universal** version of Simulator.
> 
> 1. Remove the existing iOS platform runtime:
> 
> ```
> sudo rm -rf "/Library/Developer/CoreSimulator/Profiles/Runtimes/iOS <version>.simruntime"
> ```
> 
> _OR use `Xcode settings -> Components -> e.g. iOS 26.5 (delete this)`_
> 
> 2. Install universal version instead:
> 
> ```
> xcodebuild -downloadPlatform iOS -architectureVariant universal
> ```
> 
> 3. Restart your Mac or at least try this:
> 
> ```
> killall Simulator 2>/dev/null 
> ```
> 
> Note that using this Rosetta based universal version degrades the Simulator performance.

yes, i've actually been using Rosetta since opening this ticket.

**Comment by alex-pan-invos** [comment]
+1

**Comment by MADrickx** [comment]
+1

**Comment by iwater** [comment]
<img width="470" height="994" alt="Image" src="https://github.com/user-attachments/assets/37429269-a4a4-4e41-9b47-cb31fdb81fd5" />

Hi everyone,

Since we've been waiting for years without an official solution from Google, I've built a native workaround to support running MLKit on Apple Silicon (arm64) iOS Simulators at full speed(without Rosetta).

### How it works
I downloaded Google's official MLKit binary distributions, repackaged them into fully-compliant `.xcframework` bundles (including the proper simulator slices), patched the binary load commands to target the simulator platform, resolved the static duplicate symbol conflicts, and published them to a custom CocoaPods Spec Repository.

### How to use
The integration is extremely straightforward. You only need to **add our private Spec source** at the top of your `Podfile` and enable **`use_frameworks! :linkage => :static`** for your target. No complex `post_install` patching scripts are required.

All the detailed patching scripts, automated workflows, and setup guides are documented in the repository. Feel free to check it out if you're interested:

👉 **GitHub Repository**: [google-mlkit-ios-arm64-simulator](https://github.com/iwater/google-mlkit-ios-arm64-simulator)

Hope this helps get your local simulator builds and CI pipelines running natively!

**Comment by jsg2021** [comment]
do you have an expo plugin to update the podfile?

**Comment by carl-sauter** [comment]
+1

**Comment by patreu22** [comment]
+1

**Comment by alex-pan-invos** [comment]
+1

**Comment by duszmox** [comment]
+1

**Comment by edwinhilgersom** [comment]
+1

**Comment by tomas-dorda** [comment]
+1
